src/Files.Shared/Extensions/LinqExtensions.cs

```diff
@@ -41,7 +41,7 @@ public static class LinqExtensions
 			return defaultValue;
 		}
 
-		public static Task<TValue?> GetAsync<TKey, TValue>(this IDictionary<TKey, Task<TValue?>> dictionary, TKey key, Func<Task<TValue?>> defaultValueFunc)
+		public static Task<TValue?> GetAsync<TKey, TValue>(this IDictionary<TKey, Task<TValue?>> dictionary, TKey key, Func<Task<TValue?>> defaultValueFunc) where TKey : notnull
 		{
 			if (dictionary is null || key is null)
```

Comment:
Does the `key is null` in the condition line 46 still makes sense with the `where TKey : notnull` ?

Reply:
`where TKey : notnull` doesn't enforce non-nullable unless we treat all nullable related warnings as errors.